libs/utils/src/rotation.ts

```diff
@@ -0,0 +1,46 @@
+import { Candidate, CandidateContest, CandidateVote } from '@votingworks/types';
+
+export function rotateArrayByLength<T>(
+  array: readonly T[],
+  length: number
+): T[] {
+  const a = [...array];
+  return a.concat(a.splice(0, (length < 0 ? 0 : length) % a.length));
+}
+
+export function getContestCandidatesInRotatedOrder({
+  contest,
+  precinctIndex,
+}: {
+  contest: CandidateContest;
+  precinctIndex: number;
+}): readonly Candidate[] {
+  if (contest.rotation?.type === 'candidateShiftByPrecinctIndex') {
+    return rotateArrayByLength(contest.candidates, precinctIndex);
+  }
+  return contest.candidates;
+}
+
+export function getContestVoteInRotatedOrder({
+  contest,
+  vote,
+  precinctIndex,
+}: {
+  contest: CandidateContest;
+  vote: CandidateVote;
+  precinctIndex: number;
+}): CandidateVote {
+  let sortedCandidates = contest.candidates;
+  if (contest.rotation?.type === 'candidateShiftByPrecinctIndex') {
+    sortedCandidates = rotateArrayByLength(contest.candidates, precinctIndex);
+  }
```

Comment:
Just use the helper above rather than duplicating it?

```suggestion
  const sortedCandidates = getContestCandidatesInRotatedOrder({
    contest,
    precinctIndex,
  });
```